backends/gdx-backend-headless/src/com/badlogic/gdx/backends/headless/ZipResourceFile.java

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (C) 2012 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.badlogic.gdx.backends.headless;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Vector;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public class ZipResourceFile {
```

Comment:
Hmm, not super keen on copying this, and then duplicating it in multiple backend.

Reply:
right, It seemed that this one was pretty much unchanged, but will refactor them properly, thanks.

Reply:
I guess what I am getting at is I don't understand why it is necessary to have this class at all. The Android team created it to solve an Android problem, it's not required in order to read things from a zip file.

Reply:
Yeah, now that I read into it it's clear this wasn't needed, I'll keep at it making it possible to read zip files on non android backends.